tsdb/querier.go

```diff
@@ -584,15 +584,25 @@ func (p *populateWithDelGenericSeriesIterator) reset(blockID ulid.ULID, cr Chunk
 	p.currChkMeta = chunks.Meta{}
 }
 
-func (p *populateWithDelGenericSeriesIterator) next() bool {
+func (p *populateWithDelGenericSeriesIterator) next(forChunks bool) bool {
```

Comment:
I dislike passing control flags into functions.
Moreover, this flag describes what the caller is intending to do; it would be better to describe how this function will behave differently.